core/types/types.cc

```diff
@@ -846,14 +851,7 @@ core::SymbolRef Types::getRepresentedClass(core::Context ctx, const core::Type *
         if (at == nullptr) {
             return core::Symbols::noSymbol();
         }
-        // If this class is a "real" generic, leave it alone. We're not quite
-        // sure yet what it means to move between the "attached" and "singleton"
-        // levels for generic singletons. However, if the singleton has kind `*`
-        // due to all type members being fixed:, we're OK to treat the
-        // AppliedType and the Symbol as interchangeable here.
-        if (at->klass.data(ctx)->typeArity(ctx) != 0) {
-            return core::Symbols::noSymbol();
-        }
```

Comment:
Is this just because the special case for things like

```ruby
class Opus::Enum
  Elem = type_template
end

class MyEnum < Opus::Enum
  Elem = type_template(fixed: self)
end

x.is_a?(Opus::Enum)
x.is_a?(MyEnum)
```

is not needed after the changes on this branch?

Reply:
Yes, that's correct. The defaulting rules in `externalType` handle type parameters a bit better now, so there's no reason to only allow types that have fixed parameters through.